shell/platform/darwin/ios/framework/Source/FlutterPlatformViewsController.h

```diff
@@ -0,0 +1,144 @@
+///
+/// Additionally, reverts the composition order to its original state at the beginning of the
+/// frame.
+- (void)cancelFrame;
+
+/// @brief Record a platform view in the layer tree to be rendered, along with the positioning and
+///        mutator parameters.
+///
+/// Called from the raster thread.
+- (void)prerollCompositeEmbeddedView:(int64_t)viewId
+                          withParams:(std::unique_ptr<flutter::EmbeddedViewParams>)params;
+
+/// @brief Returns the`FlutterTouchInterceptingView` with the provided view_id.
+///
+/// Returns nil if there is no platform view with the provided id. Called
+/// from the platform thread.
+- (FlutterTouchInterceptingView*)flutterTouchInterceptingViewForId:(int64_t)viewId;
+
+/// @brief Determine if thread merging is required after prerolling platform views.
+///
+/// Called from the raster thread.
+- (flutter::PostPrerollResult)postPrerollActionWithThreadMerger:
+                                  (const fml::RefPtr<fml::RasterThreadMerger>&)rasterThreadMerger
+                                                impellerEnabled:(BOOL)impellerEnabled;
+
+/// @brief Mark the end of a compositor frame.
+///
+/// May determine changes are required to the thread merging state.
+/// Called from the raster thread.
+- (void)endFrameWithResubmit:(BOOL)shouldResubmitFrame
+                threadMerger:(const fml::RefPtr<fml::RasterThreadMerger>&)rasterThreadMerger
+             impellerEnabled:(BOOL)impellerEnabled;
+
+/// @brief Returns the Canvas for the overlay slice for the given platform view.
+///
+/// Called from the raster thread.
+- (flutter::DlCanvas*)compositeEmbeddedViewWithId:(int64_t)viewId;
+
+/// @brief Discards all platform views instances and auxiliary resources.
+///
+/// Called from the raster thread.
+- (void)reset;
+
+/// @brief Encode rendering for the Flutter overlay views and queue up perform platform view
+/// mutations.
+///
+/// Called from the raster thread.
+- (BOOL)submitFrame:(std::unique_ptr<flutter::SurfaceFrame>)frame
+     withIosContext:(const std::shared_ptr<flutter::IOSContext>&)iosContext
+          grContext:(GrDirectContext*)grContext;
+
+/// @brief Handler for platform view message channels.
+- (void)onMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result;
+
+/// @brief Returns the platform view id if the platform view (or any of its descendant view) is
+/// the first responder.
+///
+/// Returns -1 if no such platform view is found.
+- (long)firstResponderPlatformViewId;
+
+/// @brief Pushes backdrop filter mutation to the mutator stack of each visited platform view.
+- (void)pushFilterToVisitedPlatformViews:(const std::shared_ptr<flutter::DlImageFilter>&)filter
+                                withRect:(const SkRect&)filterRect;
+
+/// @brief Pushes the view id of a visted platform view to the list of visied platform views.
+- (void)pushVisitedPlatformViewId:(int64_t)viewId;
+
+@end
+
+@interface FlutterPlatformViewsController (Testing)
```

Comment:
If we feel like, we could move this to a separate header, but this header is already internal so not particularly necessary/urgent.